pkg/cmd/provisioning/gcp/create_service_accounts.go

```diff
@@ -0,0 +1,350 @@
+		}
+		// commands to add bindings for workload identity user role to service account
+		for _, identityPoolBindingName := range identityProviderBindingNames {
+			addPolicyBindingCmds = append(addPolicyBindingCmds, fmt.Sprintf(addPolicyBindingForSvcAcctCmd, identityPoolBindingName, workloadIdentityUserRole))
+		}
+		addPolicyBindingScript := createShellScript(addPolicyBindingCmds)
+		addPolicyBindingScriptName := fmt.Sprintf(provisioning.AddIAMPolicyBindingScriptName, serviceAccountNum, serviceAccountName)
+		addPolicyBindingScriptFullPath := filepath.Join(targetDir, addPolicyBindingScriptName)
+		if err := provisioning.SaveToFile(fmt.Sprintf("script to add policy bindings for service account %s", serviceAccountName), addPolicyBindingScriptFullPath, []byte(addPolicyBindingScript)); err != nil {
+			return "", errors.Wrap(err, "Failed to save 'add policy binding' shell script")
+		}
+
+		// Create shell script to create client credentials configuration files that operators pods can use to impersonate
+		// the IAM service account
+		credentialsConfigFilePath := filepath.Join(targetDir, "credentials_configurations", fmt.Sprintf("%s_credentials_config.json", serviceAccountName))
+		generateCredentialsConfigScript := createShellScript([]string{
+			fmt.Sprintf(generateCredentialsConfigCmd, project, workloadIdentityPool, workloadIdentityProvider, credentialsConfigFilePath),
+		})
+		generateCredentialsConfigScriptName := fmt.Sprintf(provisioning.GenerateCredentialsConfigScriptName, serviceAccountNum, serviceAccountName)
+		generateCredentialsConfigScriptFullPath := filepath.Join(targetDir, generateCredentialsConfigScriptName)
+		if err := provisioning.SaveToFile(fmt.Sprintf("script to generate credentials config for service account %s", serviceAccountName), generateCredentialsConfigScriptFullPath, []byte(generateCredentialsConfigScript)); err != nil {
+			return "", errors.Wrap(err, "Failed to save 'generate credentials config' shell script")
+		}
+
+		// secrets are not populated with credentials in generate mode, you need to create client credentials config
+		// using 'gcloud iam workload-identity-pools create-cred-config' command, base64 encode resulting json and
+		// populate service_account.json field in the secret manifests
+		encodedCredentialsConfig = ""
+		if err := writeCredReqSecret(credReq, targetDir, encodedCredentialsConfig, generateCredentialsConfigScriptFullPath); err != nil {
+			return "", errors.Wrap(err, "Failed to save Secret for install manifests")
+		}
+
+		return "", nil
+
+	default:
+		var serviceAccount *iamadminpb.ServiceAccount
+		serviceAccount, err = actuator.GetServiceAccount(client, serviceAccountID)
+		if err != nil {
+			if status.Code(err) == codes.NotFound {
+				serviceAccount, err = actuator.CreateServiceAccount(client, serviceAccountID, serviceAccountName, provisioning.CreatedByCcoctl, project)
```

Comment:
can we put a long description containing the OpenShift ServiceAccount name (the shortened GCP ServiceAccount names are pretty truncated)

Reply:
Currently, description for gcp service account is `Created by OpenShift ccoctl`. Should we make it `Created by OpenShift ccoctl, OpenShift ServiceAccount name is <openshift-sa-name> ` ?

Reply:
^ that or something like `Created by ccoctl for ServiceAccount <openshift-sa-name>` (if what you proposed is too long)